plv8.cc

```diff
@@ -1620,7 +1620,8 @@ GetGlobalContext(Persistent<Context>& global_context)
 				fake_fcinfo.arg[1] = CStringGetDatum(arg);
 				Datum ret = has_function_privilege_id(&fake_fcinfo);
 #else
-				MemSet(fake_fcinfo, 0, sizeof(fake_fcinfo));
+				fake_fcinfo = (FunctionCallInfo) palloc(SizeForFunctionCallInfo(2));
```

Comment:
here you're allocating over an existing object.

Reply:
also missing a `pfree()` so causing a memory leak

Reply:
I can believe that. (Though I don't see how. There's a declaration `FunctionCallInfo fake_fcinfo;` on line 1601, then this is the first statement that refers to fake_fcinfo following the declaration.)

Without the allocation, I get a segfault at the line `MemSet(fake_fcinfo, 0, sizeof(fake_fcinfo));` I'm interpreting that as an attempt to zero out unallocated memory.

Backtrace from a core file:

```
Core was generated by `postgres: 12/main: import import ::1(42918) SELECT                            '.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fdeeb657a2c in GetGlobalContext (global_context=...) at plv8.cc:1625
1625					fake_fcinfo->flinfo = &flinfo;
[Current thread is 1 (Thread 0x7fdf710b8a40 (LWP 1747400))]
(gdb) bt
#0  0x00007fdeeb657a2c in GetGlobalContext (global_context=...) at plv8.cc:1625
```

Which shows in syslog as
```
Dec 16 17:01:39 behenian kernel: [1559398.124502] postgres[1747400]: segfault at 0 ip 00007fdeeb657a2c sp 00007ffc33f6d320 error 6 in plv8-3.0alpha.so[7fdeeb64a000+934000]
Dec 16 17:01:39 behenian kernel: [1559398.124517] Code: 1f 80 00 00 00 00 48 83 c0 08 48 c7 40 f8 00 00 00 00 48 39 d0 75 ef 48 8b 05 c8 e3 b9 00 31 ff c7 85 d8 fe ff ff 00 00 00 00 <48> 89 0f 48 8b 00 48 89 85 f0 fe ff ff b8 02 00 00 00 66 89 47 1e
```

Reply:
likely what occurred is that Postgres internals changed the `typedef` for FunctionCallInfo.  I'll take a deeper look into it when I get the time.

Reply:
if that's the case, then it will need to be `#if`'d to work correctly across all postgres versions that plv8 supports.

Reply:
Thank you for your feedback. 

* You're right about the changed typedef for FunctionCallInfo. You set up the  `#if`s a few months ago in commit 77b941128cdb; I'm building off that.

* To avoid memory leaks, I've changed fake_fcinfo to be stack-allocated, instead of using palloc(). The LOCAL_FCINFO macro is from the PG12 version of fmgr.h.